### PR TITLE
Migrate from YARP_telemetry to robometry

### PR DESCRIPTION
**Pull request by Nicogene**
See https://github.com/robotology/robometry/pull/173

Please review code.

### CODE REVIEW

**Comment by Nicogene** [comment]
I am not sure if the changes in `ci.yml` works

File: wrappers/IWearLogger/src/IWearLogger.cpp [hunk 1]
```diff
@@ -160,8 +160,8 @@ class IWearLogger::impl
     yarp::dev::IPreciselyTimed* iPreciselyTimed = nullptr;
     std::mutex loggerMutex;
     IWearLoggerSettings settings;
-    yarp::telemetry::experimental::BufferConfig bufferConfig;
-    yarp::telemetry::experimental::BufferManager<double> bufferManager;
+    robometry::BufferConfig bufferConfig;
+    robometry::BufferManager<double> bufferManager;
```
Comment (S-Dafarra) [inline wrappers/IWearLogger/src/IWearLogger.cpp:164]: I would suggest removing the template

**Review by lrapetti — COMMENTED** [review]
Overall looks good, but the CI seems to be failing on WIndows

**Comment by lrapetti** [comment]
Btw, it might be worth mentioning this modification in the [`CHANGELOG.md`](https://github.com/robotology/wearables/blob/master/CHANGELOG.md)

**Comment by Nicogene** [comment]
> Overall looks good, but the CI seems to be failing on WIndows

I think that the CI will be broken until the new robometry packets will be released am I right @traversaro ?

**Comment by traversaro** [comment]
> > Overall looks good, but the CI seems to be failing on WIndows
> 
> I think that the CI will be broken until the new robometry packets will be released am I right @traversaro ?

Yes

**Comment by traversaro** [comment]
> Here [9caf780](https://github.com/robotology/wearables/commit/9caf780e949891b7890a15ab387f38ff7e4a7f46) I removed the installation of robometry in the CI for windows since the conda packets.
> 
> This means that IWearLogger will be untested under windows until we have the packets What do you think about @lrapetti @traversaro ?

I think it is a good idea.

**Comment by lrapetti** [comment]
> This means that IWearLogger will be untested under windows until we have the packets
> What do you think about @lrapetti @traversaro ?

That's ok. We could open an issue after merging this PR in order to remember to add it back


@Nicogene if the PR is ready I would proceed merging.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -146,10 +146,10 @@ jobs:
           if: steps.cache-source-deps.outputs.cache-hit != 'true' && contains(matrix.os, 'windows')
           shell: bash -l {0}
           run: |
-            # Compilation related dependencies 
+            # Compilation related dependencies
             mamba install cmake compilers make ninja pkg-config
             # Actual dependencies
-            mamba install -c robotology yarp wearables idyntree matio-cpp yarp-telemetry
+            mamba install -c robotology yarp wearables idyntree matio-cpp
```
Comment (lrapetti) [inline .github/workflows/ci.yml:152]: Not related to this PR, but `wearables` should not be here

File: wrappers/IWearLogger/src/IWearLogger.cpp (hunk 1 above)
Comment (lrapetti) [reply wrappers/IWearLogger/src/IWearLogger.cpp]: Accordingly to https://github.com/robotology/robometry/pull/175 we should now be able to remove the template, am I right?

Comment (Nicogene) [reply wrappers/IWearLogger/src/IWearLogger.cpp]: Yes exactly, I removed it

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Nicogene) [reply .github/workflows/ci.yml]: I did it here https://github.com/robotology/wearables/pull/155/commits/7190c9ed7848f311d6b26474ba644a99e4f9a30f

**Pull request merged**